build.gradle

```diff
@@ -61,27 +66,31 @@ subprojects {
 
     configurations.all {
         resolutionStrategy {
+            dependencySubstitution {
+                // The substitution is needed to get ASM 5.2 whilst Kryo still depends on 5.0.4
```

Comment:
Does a newer Kryo help?

Reply:
Kryo 4.0.2 _is_ the latest version, unless you want to use a release candidate.

One of the "gotchas" about OSGi is that dependency version ranges are fixed by the metadata in the manifest and cannot be upgraded externally. I had to bash on Gradle quite hard so that published POM aligned with the manifest, as otherwise the OSGi "Resolve" task needed extra help.

Reply:
fair enough